client/java/src/main/java/io/openlineage/client/utils/jdbc/JdbcDatasetUtils.java

```diff
@@ -61,6 +63,10 @@ public static DatasetIdentifier getDatasetIdentifier(
     try {
       JdbcExtractor extractor = getExtractor(uri);
       JdbcLocation location = extractor.extract(uri, properties);
+      GravitinoInfoProviderImpl provider = GravitinoInfoProviderImpl.getInstance();
+      if (provider.useGravitinoIdentifier()) {
+        return GravitinoUtils.getGravitinoDatasetIdentifierFromJdbc(location.toName(parts));
```

Comment:
Why this approach is used instead of `GravitinoExtractor implements JdbcExtractor`?